File: typescript/src/models/webhookOrderStatus.ts
// this file is @generated

export enum WebhookOrderStatus {
  Created = "created",
  ManualReview = "manual_review",
  Confirmed = "confirmed",
  Canceled = "canceled",
  Shipped = "shipped",
  Fulfilled = "fulfilled",
}

export const WebhookOrderStatusSerializer = {
  _fromJsonObject(object: any): WebhookOrderStatus {
    return object;
  },

  _toJsonObject(self: WebhookOrderStatus): any {
    return self;
  },
};

File: rust/src/models/webhook_order_status.rs
// this file is @generated
use std::fmt;

use serde::{Deserialize, Serialize};

#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
pub enum WebhookOrderStatus {
    #[serde(rename = "created")]
    Created,
    #[serde(rename = "manual_review")]
    ManualReview,
    #[serde(rename = "confirmed")]
    Confirmed,
    #[serde(rename = "canceled")]
    Canceled,
    #[serde(rename = "shipped")]
    Shipped,
    #[serde(rename = "fulfilled")]
    Fulfilled,
}

impl fmt::Display for WebhookOrderStatus {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let value = match self {
            Self::Created => "created",
            Self::ManualReview => "manual_review",
            Self::Confirmed => "confirmed",
            Self::Canceled => "canceled",
            Self::Shipped => "shipped",
            Self::Fulfilled => "fulfilled",
        };
        f.write_str(value)
    }
}
